src/main/java/bio/terra/service/snapshotbuilder/utils/SearchConceptsQueryBuilder.java

```diff
@@ -19,21 +19,56 @@
 
 public class SearchConceptsQueryBuilder {
 
-  private SearchConceptsQueryBuilder() {}
-
-  public static Query buildSearchConceptsQuery(
+  public static final String CONCEPT_ID = "concept_id";
+  public static final String CONCEPT = "concept";
+  public static final String CONCEPT_ANCESTOR = "concept_ancestor";
+  public static final String CONCEPT_NAME = "concept_name";
+  public static final String CONCEPT_CODE = "concept_code";
+  public final String ANCESTOR_CONCEPT_ID = "ancestor_concept_id";
+  public final String DESCENDANT_CONCEPT_ID = "descendant_concept_id";
+
+  /**
+   * Generate a query that retrieves all the concepts from the given searched text. If a search text
+   * is not provided, a search will be made only on the domain.
+   *
+   * <p>GCP: SELECT c.concept_name, c_concept_id, COUNT(DISTINCT co.person_id) AS count FROM
+   * `concept` AS c JOIN `concept_ancestor` AS c0 ON c0.ancestor_concept_id = c.concept_id LEFT JOIN
+   * `'domain'_occurrence` AS co ON co.'domain'_concept_id = c0.descendant_concept_id WHERE
+   * (c.domain_id = 'domain' AND (CONTAINS_SUBSTR(c.concept_name, 'search_text') OR
+   * CONTAINS_SUBSTR(c.concept_code, 'search_text'))) GROUP BY c.concept_name, c.concept_id ORDER BY
+   * count DESC LIMIT 100
+   *
+   * <p>AZURE: SELECT c.concept_name, c_concept_id, COUNT(DISTINCT co.person_id) AS count FROM
+   * `concept` AS c JOIN `concept_ancestor` AS c0 ON c0.ancestor_concept_id = c.concept_id LEFT JOIN
+   * `'domain'_occurrence` AS co ON co.'domain'_concept_id = c0.descendant_concept_id WHERE
+   * (c.domain_id = 'domain' AND (CHARINDEX('search_text', c.concept_name) > 0 OR
+   * CHARINDEX('search_text', c.concept_code) > 0)) GROUP BY c.concept_name, c.concept_id ORDER BY
+   * count DESC LIMIT 100
+   */
+  public Query buildSearchConceptsQuery(
       SnapshotBuilderDomainOption domainOption, String searchText) {
-    var conceptTablePointer = TablePointer.fromTableName("concept");
+    var conceptTablePointer = TablePointer.fromTableName(CONCEPT);
+    var conceptAncestorPointer = TablePointer.fromTableName(CONCEPT_ANCESTOR);
     var domainOccurrencePointer = TablePointer.fromTableName(domainOption.getTableName());
     var conceptTableVariable = TableVariable.forPrimary(conceptTablePointer);
-    var nameField = conceptTableVariable.makeFieldVariable("concept_name");
-    var idField = conceptTableVariable.makeFieldVariable("concept_id");
+    var nameField = conceptTableVariable.makeFieldVariable(CONCEPT_NAME);
+    var idField = conceptTableVariable.makeFieldVariable(CONCEPT_ID);
+
+    // FROM 'concept' as c
+    // JOIN concept_ancestor as c0 ON c0.ancestor_concept_id = c.concept_id
+    var conceptAncestorTableVariable =
+        TableVariable.forJoined(conceptAncestorPointer, ANCESTOR_CONCEPT_ID, idField);
```

Comment:
I think we had discussed inlining the `pointer`, since it is only used here.

I would also encourage using the types, rather than `var`, and pulling `TableVariable` or `Field` from the variable names, since that's redundant with the type information and makes the variables more verbose. (You could just have, for example, `conceptAncestor`, and then it would read more cleanly below where we make the field variables)